devel-up

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/python3
+
+'''
+    Tiny convenient and *safe* wrapper around `oc process/apply` for devel
+    pipelines. Can be run multiple times; subsequent runs will replace existing
+    resources.
+
+    Example usage:
+        ./devel-up --update \
+            --pipeline https://github.com/jlebon/fedora-coreos-pipeline \
+            --config https://github.com/jlebon/fedora-coreos-config@wip \
+            --cosa-img quay.io/jlebon/coreos-assembler:master
+            # OR
+            --cosa https://github.com/jlebon/coreos-assembler
+
+    Deleting devel pipeline:
+        ./devel-up --delete
+'''
+
+import os
+import sys
+import json
+import argparse
+import subprocess
+
+
+def main():
+    args = parse_args()
+    resources = process_template(args)
+
+    if args.update:
+        update_resources(args, resources)
+
+        if args.start:
+            print("Starting:")
+            out = subprocess.run(['oc', 'start-build',
+                                  # hack the name; should extract from manifest
+                                  f'{args.prefix}fedora-coreos-pipeline'],
+                                 check=True, stdout=subprocess.PIPE,
+                                 encoding='utf-8')
+            print(f"  {out.stdout.strip()}")
+            print()
+        else:
+            print("You may start your devel pipeline with:")
+            print(f"  oc start-build {args.prefix}fedora-coreos-pipeline")
+            print()
+    else:
+        assert args.delete
+
+        delete_resources(args, resources)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    action = parser.add_mutually_exclusive_group(required=True)
+    action.add_argument("--update", action='store_true',
+                        help="Create or update devel pipeline")
+    action.add_argument("--delete", action='store_true',
+                        help="Delete devel pipeline")
+    parser.add_argument("--start", action='store_true',
+                        help="Start build after updating it")
+    parser.add_argument("--prefix", help="Devel prefix to use for resources",
+                        default=get_username())
+    parser.add_argument("--pipeline", metavar='<URL>[@REF]',
```

Comment:
is `@REF` a convention? I know at least `oc new-build` uses `#`:

```
  # Create a build config from a remote repository using its beta2 branch
  oc new-build https://github.com/openshift/ruby-hello-world#beta2
```

Reply:
Inspired by ci-operator: https://github.com/openshift/ci-operator/#usage. I like that `@` actually means "at" which is appropriate here. I'm open to switching it though.

Reply:
ok - yeah I didn't know where it came from. I also don't know if there is a case where we'll want to delineate between branch or commit hash. In that case we could have used `#` for branch and `@` for commit hash. It's cool to leave it like it is.